### ISSUE TITLE
JDK 15 as required in 16.0.0 and 15.0.1-SNAPSHOT release version for 16.

### ISSUE THREAD

**Issue by NickKononovich** [googleads/google-ads-java#507]
Hi team. I checked tags and seems 16.0.0 was released with 15.0.1-SNAPSHOT of Google Ads version.
And also and most important, I have JDK 11 and while building the project I have - java.lang.IllegalArgumentException: Unsupported class file major version 59. Could you please tell if 16.0.0 is supported starting from JDK 15? Thanks.

**Comment by jradcliff** [comment]
Hi Nick,

Do you have the same problems if you use Gradle instead of Maven? I realize there are some issues with the poms as released, but hopefully using Gradle will work for you while we fix this.

Also, if you only want to use the library as a dependency, you shouldn't have to compile it locally. Setting your dependency on our library to `com.google.api-ads:google-ads:16.0.0` should suffice.

It appears the `.class` files in the JARs are for Java 1.8 (as expected), not JDK15. For example, I downloaded google-ads-16.0.0.jar from the central Maven repo, extracted it, and checked using `files`:

```
$ file com/google/ads/googleads/v7/utils/ResourceNames.class
com/google/ads/googleads/v7/utils/ResourceNames.class: compiled Java class data, version 52.0 (Java 1.8)
```

Thanks,
Josh

**Comment by NickKononovich** [comment]
Hi Josh, Thank you for feedback. 
If I check version of this file 
`javap -verbose META-INF/versions/15/org/bouncycastle/jcajce/provider/asymmetric/edec/BC11XDHPrivateKey.class | grep "major"`
  major version: 59
Could you please check this files?

**Comment by nwbirnie** [comment]
I tried to reproduce the issue by creating a new maven project, depending on `google-ads` and creating a new instance of `GoogleAdsClient`. I tried compiling this with JDK 8 and JDK 11, but [didn't get any compiler errors](https://gist.github.com/nwbirnie/a95f9692c2b6aa486bf7ac2562a86dad).

I also looked at the jar containing `META-INF/versions/15/org/bouncycastle/jcajce/provider/asymmetric/edec/BC11XDHPrivateKey.class`. This is found in [`bcprov-jdk15on-1.67.jar`](https://github.com/googleads/google-ads-java/blob/main/third_party/bcprov-jdk15on-1.67.jar). This is a dependency of gRPC, specifically `grpc-xds`.

![image](https://user-images.githubusercontent.com/410275/143039772-9f7aaeaa-3016-4dc4-ae13-9b20593887b9.png)

To the best of my knowledge there is no reason that we have to depend on [xDS](https://www.envoyproxy.io/) (it's an edge proxy and service discovery protocol, used in service meshes and the like). It like we didn't depend on `gprc-xds` at the [last release](https://gist.github.com/nwbirnie/2751410510fea258a81d52340cb8b54c), so it would make sense that if this is somehow broken that it would break on the bump to `16.0.0`. The jar itself is compiled as [multi-version jar](https://docs.oracle.com/javase/10/docs/specs/jar/jar.html#multi-release-jar-files), it has `META-INF/versions/11` and `META-INF/versions/9` indicating that both JDK 9 and 11 are supported. The main classes directory doesn't contain this class, however it does have some [similar entries](https://gist.github.com/nwbirnie/a274bdbdb8fd7d81fd3e4f9b8f9c3742) for JDK 5.

So I think we could remove our over-zealous dependency on `grpc-all` in favour of `grpc-api` and maybe `grpc-stub` and some of the others which are probably already runtime dependencies of `gax-grpc`. In terms of a workaround, @NickKononovich I think you could add a dependency exclusion to remove `grpc-xds` from your project's dependencies. I'd be curious if you could provide a reliable way of reproducing your issue, since it's not clear to me why this breaks at compile time for you.

**Comment by nwbirnie** [comment]
I just realized after posting this that the jar name for bcprov is jdk15on suggesting that it is compiled for JDK 15+. This is incorrect, it's intended to be JDK 1.5+.

**Comment by NickKononovich** [comment]
Hi @nwbirnie. 
I think our build tool (https://buck.build/) doesn't support multi-jars out of the box. That is why it tried to read classes for higher JDK. Now we found the workaround. However, your fix with simplifying dependency tree is also nice and I want to test once you release. 
Thank you for help here. 

With Best Regards.

**Comment by nwbirnie** [comment]
Oh interesting, do you work for Facebook by any chance? I've merged the PR, but can't release right now due to holiday freeze. Will get this out next week.

**Comment by nwbirnie** [comment]
Just released 16.0.1 which should resolve this issue. Please let me know if you still run into this issue.
